web/modules/custom/dpl_related_content/src/Services/RelatedContent.php

```diff
@@ -0,0 +1,371 @@
+<?php
+
+namespace Drupal\dpl_related_content\Services;
+
+use Drupal\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Database\Database;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+
+/**
+ * Load related nodes and events, based on filters.
+ */
+class RelatedContent {
+
+  /**
+   * The entity type interface.
+   */
+  protected EntityTypeManagerInterface $entityTypeManager;
+
+  /**
+   * The minimum of items that must be in the slider.
+   */
+  private int $minItems = 4;
+
+  /**
+   * How many items we max display in the slider.
+   */
+  private int $maxItems = 16;
+
+  /**
+   * The field on nodes, to sort by.
+   */
+  private string $nodeSortField = 'created';
+
+  /**
+   * The node bundles that should show up in the results.
+   *
+   * @var array|string[]
+   *  List of node bundles - e.g. articles.
+   */
+  private array $nodeBundles = ['article'];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+  ) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): static {
+    return new static(
+      $container->get('entity_type.manager'),
+    );
+  }
```

Comment:
Please delete this function.

It is not used anywhere. In your `services.yml` you declare what is passed to the constructor. This function is only relevant for classes that implement [`ContainerInjectionInterface`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21DependencyInjection%21ContainerInjectionInterface.php/interface/ContainerInjectionInterface/9).